### DISCUSSION TITLE
method number_of_edges

### DISCUSSION THREAD

**Discussion by Quasarrs** [networkx/networkx#7442] [Q&A]
`import` networkx as nx
G = nx.Graph()
G.add_edge(0, 1)
G.add_edge(1, 2)
print(G.number_of_edges(0, 2))
0`

Tell me why the method does not want to return 2 when I built a graph of 3 connected nodes?
Between nodes 0 and 2 there are 2 edges.
If I'm wrong, suggest a method that would count the number of edges between nodes

**Comment by Quasarrs** [comment]
> `import` networkx as nx G = nx.Graph() G.add_edge(0, 1) G.add_edge(1, 2) print(G.number_of_edges(0, 2)) 0`
> 
> Tell me why the method does not want to return 2 when I built a graph of 3 connected nodes? Between nodes 0 and 2 there are 2 edges. If I'm wrong, suggest a method that would count the number of edges between nodes

Seems to have resolved the issue with
nx.all_simple_edge_paths

**Reply by rossbar** [reply]
Yes - `number_of_edges` reports the total number of edges present in the graph, not the number of edges between two nodes.

Generally speaking, it's always good to review the documentation of functions/methods to see what they do and how to call them correctly. In Jupyter/IPython, you can do this with `?`, i.e. `G.number_of_edges?`. In "vanilla" Python you can use the builtin `help` function: `help(G.number_of_edges)`.

For a more comprehensive introduction to the library, check out the [tutorial](https://networkx.org/documentation/latest/tutorial.html)
